projects/showcase/cypress/support/e2e-driver.ts

```diff
@@ -1,198 +1,112 @@
-      this.compareToReference().should('eq', 0);
-      return this;
-    },
-    isNotSameAsSnapshot: (): E2eDriver => {
-      this.compareToReference().should('be.gt', 0);
-      return this;
-    },
-    mapCanvasExists: (): E2eDriver => {
-      cy.get('canvas.maplibregl-canvas').should('have.length', 1);
-      return this;
-    },
-    mapObjectLoaded: (): E2eDriver => {
-      cy.get('[data-loaded="true"]').should('exist');
-      return this;
-    },
-    mapTerrainButtonIsOff: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain').should('exist');
-      return this;
-    },
-    mapTerrainButtonIsOn: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain-enabled').should('exist');
-      return this;
-    },
-    helloWorldPopupExists: (): E2eDriver => {
-      cy.get('.maplibregl-popup').should('exist');
-      cy.get('.custom-popup-class1').should('exist');
-      cy.get('.custom-popup-class2').should('exist');
-      cy.get('.maplibregl-popup-content').within(() => {
-        cy.get('div').should('have.text', 'Hello world !');
-      });
-      return this;
-    },
-    helloWorldPopupDoesNotExist: (): E2eDriver => {
-      cy.get('.maplibregl-popup').should('not.exist');
-      return this;
-    },
-    customHelloButtonExists: (): E2eDriver => {
-      cy.get('.custom-control').should('have.text', ' Hello ');
-      return this;
-    },
-    customHelloButtonDoesNotExist: (): E2eDriver => {
-      cy.get('.custom-control').should('not.exist');
-      return this;
-    },
-    fullscreenControlExists: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-fullscreen').should('exist');
-      return this;
-    },
-    fullscreenControlDoesNotExist: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-fullscreen').should('not.exist');
-      return this;
-    },
-    customPopupContainsHello: (): E2eDriver => {
-      cy.get('@alertStub').should('have.been.calledOnceWith', 'Hello');
-      return this;
-    },
-  };
+  get = {
+    canvas: (): Cypress.Chainable<HTMLCanvasElement> =>
+      this.helper.get
+        .element("canvas.maplibregl-canvas", 0)
+        .its(0) as unknown as Cypress.Chainable<HTMLCanvasElement>,
 
-  private getCanvas(): Cypress.Chainable<any> {
-    return cy.get('canvas.maplibregl-canvas').its(0).should('not.be.undefined');
-  }
+    imageSnapshot: () => {
+      const snapshot = this.get
+        .canvas()
+        .then((mapCanvas: HTMLCanvasElement) => ({
+          width: mapCanvas.width,
+          height: mapCanvas.height,
+          buffer: this.get.imageBitmapBuffer(mapCanvas),
+        }));
+      this.get.canvas();
 
-  private toImageBitmapBuffer(canvas: HTMLCanvasElement) {
-    const base64 = canvas.toDataURL('image/png').replace(/data:.*;base64,/, '');
-    const buff = Cypress.Buffer.from(base64, 'base64');
-    const png = PNG.sync.read(buff as any);
-    return png.data;
-  }
+      return snapshot;
+    },
+
+    imageBitmapBuffer: (canvas: HTMLCanvasElement) => {
```

Comment:
Same here, this looks like something that should not be exposed to the test code.

Reply:
Why?

Reply:
Doesn't sound like something that would make sense to "PM" if they were writing the tests. This should be the "language" of the driver, right?